keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -473,11 +447,11 @@ spec:
     # Additional annotations or labels for the
     # ResourceClaim could be specified here.
   spec:
-    resourceClassName: "acme-gpu"
-    parametersRef:
-      apiGroup: gpu.example.com
-      kind: GPURequirements
-      name: device-consumer-gpu-parameters
+    requests:
```

Comment:
I have two arguments:
- ResourceSlice and ResourceClaim are the "outer API" that is more abstract. We use it for devices now, but this could change. How to deal with devices is the "inner API", embedded in both.
- A PersistentVolumeClaim is asking for exactly one volume. A ResourceClaim doesn't. Calling it "DeviceClaim" would give the wrong impression that it's just for "one device".

For this specific field, @klueska [had already suggested](https://github.com/kubernetes/enhancements/pull/4709#discussion_r1635314449) renaming to "devices". I was initially okay with it because it would have made sense ("inner API"), but when thinking about the implications on how to describe and reference the entries elsewhere we found that "device name" became ambiguous (entry in this slice vs. entry in ResourceSlice). "deviceRequests" might have worked, but for now we just kept "requests". See that thread for details.

Reply:
That means if we add per-device allocatable resources, we end up with `claim.spec.requests[*].device.requests` which is an improvement on `claim.spec.requests[*].requests` which is what we would get today.

I still think it should just be `DeviceClaim` though, if I had my druthers. It would be reasonable in my mind to have `DeviceClaim`, and still keep `ResourceSlice`. Later, if we have Gizmos, maybe we have a `GizmoClaim`. By that time we will understand our needs enough I can imagine we may be OK with putting some cross-claim constraints into PodSpec, if - highly highly speculatively - we need some sort of constraints between devices and gizmos. And even if we don't want that, if we really really need it, we can even have a special `CompositeClaim` or something. But this all seems like speculation on top of speculation on top of speculation...

What we know about now are devices; `DeviceClaim` constrains us to thinking about devices.

Reply:
To me, a `ResourceClaim` is still the general concept, with corresponding support in the pod API and in the resource claim controller. I'm -2 on making all of that specific to devices. Only what each "claim.spec.request" is for could vary.

With that in mind, `claim.spec.requests[*].device` makes sense to me. I'm pretty sure I had a one-of there at some point and only removed it because of "we only do devices, drop this" review feedback :sweat_smile:

Reply:
works for me

Reply:
SGTM -